en/services/ftp.md

```diff
@@ -209,46 +209,49 @@ On completion of the burst (or the file), if there are any missing parts of the
+    Note left of Server: Server streams data in chunks at increasing offsets with burst_complete=0
+    Server->>Client: ACK(session, burst_complete=0, offset=0, size=len(buffer), data[0]=buffer)
+    Server->>Client: ACK(session, burst_complete=0, offset=<current_offet>, size=len(buffer), data[0]=buffer)
+    Server->>Client: ...
+    Note left of Server: For last message in burst set burst_complete=1
+    Server->>Client: ACK(session, burst_complete=1, offset=<offset_of_last_burst_chunk>, size=len(buffer), data[0]=buffer)
+    Note over Client,Server: Read any missing parts using BurstReadFile (sequence above) or ReadFile
+    Note over Client,Server: Close session when whole file collected
+    Client->>Server:  TerminateSession(session)
+    Server->>Client: ACK()
 -->
 
 The sequence of operations is:
 1. GCS (client) sends [OpenFileRO](#OpenFileRO) command specifying the file path to open.
    - The payload must specify: `data[0]`= file path string, `size`=length of file path string.
-1. Drone (server) responds with either 
+1. Drone (server) responds with either
    - ACK on success. The [payload](#payload) must specify fields: `session` = file session id, `size` = 4, `data` = length of file that has been opened.
-   - NAK with [error information](#error_codes), e.g. `NoSessionsAvailable`, `FileExists`. 
+   - NAK with [error information](#error_codes), e.g. `NoSessionsAvailable`, `FileExists`.
      The GCS may cancel the operation, depending on the error.
 1. Client (i.e. GCS) sends [BurstReadFile](#BurstReadFile) command specifying the part of the file that it wants to get.
    - The payload must specify: `session`: the current session id, `offset` = offset in file of start of burst, `data`= length of burst, `size`=4.
-1. Server (drone) responds with either 
-   - ACK on success. The [payload](#payload) must specify fields: `session` = file session id, `size` = 4, `data` = lenght of file in burst. 
+1. Server (drone) responds with either
+   - ACK on success. The [payload](#payload) must specify fields: `session` = file session id, `size` = 4, `data` = lenght of file in burst.
    - NAK with [error information](#error_codes). The client may cancel the operation, depending on the error.
 1. Server sends stream of [BurstReadFile](#BurstReadFile) data to client (without ACK) until the whole burst is sent, or a server-defined burst size limit is reached.
-   - The payload must specify: `session`=current session, `size`=size of data to read, `offset`= position in original data of current chunk.
+   - The payload must specify: `session`=current session, `size`=size of data to read per burst (max equal to payload size), `offset`= position in original data of current chunk.
    - The payload must specify `burst_complete=0` for all chunks, except the last one, which must set `burst_complete=1`.
 1. Client repeats the `BurstReadFile` cycle at different offsets until the whole file is retrieved.
```

Comment:
Doesn't this imply that the client might send multiple requests. I don't think you'd do this because all of them will be from offset to end.

Reply:
per burst message?

Reply:
Yes, but that's not the point of the comment here (- I added the per message in line 250 above).

> Doesn't this imply that the client might send multiple requests. I don't think you'd do this because all of them will be from offset to end.

I.e. the client would never repeats the cycle at different offsets! It gets from start to end. If it got the cycle at different offsets this would be very inefficient because you'd always get the whole chunk from the current offset even if you only needed a few bytes. 

So I think 
```suggestion
```

Reply:
Right!